ducktape/mark/_parametrize.py

```diff
@@ -22,6 +22,11 @@
 
 def mark_as(fun, mark):
     """Attach a tag indicating that fun has been marked with the given mark"""
+
+    if isinstance(fun, TestGenerator):
```

Comment:
I'm not clear why we have this special case. There's already a check below for if the function has a `marks` attribute. Why the special case here? Is this because you want the `marks` `@property` to include the wrapped object's marks as well?

`isinstance` has its uses, but it's pretty unpythonic. I'm wondering if there's a better solution here. Normally I'd suggest using the `@marks.setter` annotation to add a setter in `TestGenerator`, but that won't work here since it's a collection that you want to add to instead of replace.